ui/src/app/pages/workspace/invalid-billing-banner.tsx

```diff
@@ -53,12 +196,24 @@ export const InvalidBillingBanner = fp.flow(
       Edit Workspace
     </Button>
   );
+
   return (
-    <ToastBanner
-      {...{ message, footer, onClose }}
-      title='This workspace has run out of initial credits'
-      toastType={ToastType.WARNING}
-      zIndex={500}
-    />
+    <>
+      {((isExpiringSoon && isEligibleForExtension) || isExpired) && (
```

Comment:
There is no reason to show this banner if one of these conditions is not true.

This component is only used by the breadcrumbs component. So this condition could be shifted up, but these conditions would need to be recalculated. In that case, if appropriate, breadcrumbs could calculate these values and then pass them down.

The only other usage of workspace in this component is the construction of the link associated with the  "Edit Workspace" button.